### ISSUE TITLE
Sphinx 4.0.0 crash on RepoBee project with Python < 3.8

### ISSUE THREAD

**Issue by slarse** [sphinx-doc/sphinx#9189]
**Describe the bug**
After upgrading to Sphinx 4.0.0, I'm getting a crash in in generating docs for one of my projects when using Python 3.6 and 3.7. The crash occurs in the `inspect` module of the standard library (which is called from `autodoc`) with this message:

```
ValueError: callable operator.itemgetter(1) is not supported by signature
```

**Full log file:** [sphinx-err-0or7_inn.log](https://github.com/sphinx-doc/sphinx/files/6447372/sphinx-err-0or7_inn.log)

Currently, either downgrading to 3.5.4 or using Python 3.8+ works fine as a workaround for me, so I'm reporting this mostly to make it known, it's not of critical importance to me.

**To Reproduce**

> **Note:** Requires Sphinx 4.0.0 AND Python < 3.8 to reproduce! I used pyenv to install 3.6.13, 3.7.10 and 3.8.9, where only 3.8.9 built the docs successfully.

```
$ git clone https://github.com/repobee/repobee.git
$ cd repobee
$ pip install -e .[DOCS,TEST]
$ cd docs
$ make html
# crash with Python < 3.8 and Sphinx 4.0.0
```

**Expected behavior**
I expect the docs to be buildable with Python 3.6 and 3.7 using Sphinx 4.0.0, as is the case with Sphinx 3.5.4.

**Your project**
https://github.com/repobee/repobee

**Screenshots**
N/A

**Environment info**
- OS: Arch Linux w. kernel 5.12.1
- Python version: 3.6.13/3.7.10
- Sphinx version: 4.0.0
- Sphinx extensions:
    * sphinx.ext.autodoc
    * sphinx.ext.doctest
    * sphinx.ext.intersphinx
    * sphinx.ext.todo
    * sphinx.ext.coverage
    * sphinx.ext.mathjax
    * sphinx.ext.ifconfig
    * sphinx.ext.viewcode
    * sphinx.ext.githubpages
    * sphinx.ext.napoleon
    * sphinx_autodoc_typehints
    * sphinx_rtd_theme
    * sphinxarg.ext
    * generate_package_docs (defined in  repobee/docs/_ext)

**Additional context**

- repobee/repobee#906
- https://readthedocs.org/projects/repobee/builds/13716737/

**Comment by slarse** [comment]
Let me know if there's anything more I can provide to debug this. I unfortunately just don't have the time right now to do it myself.

**Comment by tk0miya** [comment]
Thank you for reporting. The attached stacktrace is helpful to me!

**Comment by slarse** [comment]
Wow, that was a fast fix! I tried your branch in #9190 and it solved the problem for me, docs building fine again with Python 3.7.10. Thanks!
